src/main.rs

```diff
@@ -166,11 +175,22 @@ fn run_tcp(cli_config: &CliConfig, dht_sk: SecretKey, tcp_onion: TcpOnion) -> im
 
     info!("Running TCP relay on {}", cli_config.tcp_addrs.iter().format(","));
 
-    tcp_server_future.select(tcp_onion_future).map(|_| ()).map_err(|(e, _)| e)
+    Box::new(tcp_server_future
+        .join(tcp_onion_future)
```

Comment:
why `join`?

Reply:
Because if we run only TCP node here without UDP then tx part of stream will be dropped in `run_udp` function and `for_each` on rx part will be finished with success.